src/test/ensemble_metadata_test.cpp

```diff
@@ -347,3 +348,315 @@ TEST(EnsembleMetadata, OneUnavailableNode) {
+        {NodeKind::CUSTOM, "custom_node", "", std::nullopt, {{"output_numbers", "output_numbers"}}, std::nullopt, {}, library, parameters_t{{"add_value", "0.5"}, {"sub_value", "1.2"}}},
+        {NodeKind::EXIT, EXIT_NODE_NAME},
+    };
+
+    pipeline_connections_t connections;
+
+    connections["custom_node"] = {
+        {ENTRY_NODE_NAME, {{"request_input_name", "input_numbers"}}}};
+
+    connections[EXIT_NODE_NAME] = {
+        {"custom_node", {{"output_numbers", "request_output_name"}}}};
+
+    auto def = std::make_unique<PipelineDefinition>(
+        "my_new_pipeline", info, connections);
+
+    ASSERT_EQ(def->validateNodes(manager), StatusCode::OK);
+    ASSERT_EQ(def->validateForCycles(), StatusCode::OK);
+
+    tensor_map_t inputs, outputs;
+    ASSERT_EQ(def->getInputsInfo(inputs, manager), StatusCode::NODE_LIBRARY_METADATA_FAILED);
+    ASSERT_EQ(def->getOutputsInfo(outputs, manager), StatusCode::NODE_LIBRARY_METADATA_FAILED);
+}
+
+struct LibraryDemultiplexer_1 {
+    static int execute(const struct CustomNodeTensor*, int, struct CustomNodeTensor**, int*, const struct CustomNodeParam*, int) {
+        return 1;
+    }
+    static int getInputsInfo(struct CustomNodeTensorInfo** info, int* infoLength, const struct CustomNodeParam*, int) {
+        *infoLength = 2;
+        *info = (struct CustomNodeTensorInfo*)malloc(*infoLength * sizeof(struct CustomNodeTensorInfo));
+
+        (*info)[0].name = "input_A_1";
+        (*info)[0].precision = FP32;
+        (*info)[0].dimsLength = 2;
+        (*info)[0].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[0].dims[0] = 1;
+        (*info)[0].dims[1] = 1000;
+
+        (*info)[1].name = "input_B_1";
+        (*info)[1].precision = FP32;
+        (*info)[1].dimsLength = 2;
+        (*info)[1].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[1].dims[0] = 1;
+        (*info)[1].dims[1] = 400;
+        return 0;
+    }
+    static int getOutputsInfo(struct CustomNodeTensorInfo** info, int* infoLength, const struct CustomNodeParam*, int) {
+        *infoLength = 2;
+        *info = (struct CustomNodeTensorInfo*)malloc(*infoLength * sizeof(struct CustomNodeTensorInfo));
+
+        (*info)[0].name = "output_A_1";
+        (*info)[0].precision = FP32;
+        (*info)[0].dimsLength = 3;
+        (*info)[0].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[0].dims[0] = 1;
+        (*info)[0].dims[1] = 3;
+        (*info)[0].dims[2] = 10;
+
+        (*info)[1].name = "output_B_1";
+        (*info)[1].precision = FP32;
+        (*info)[1].dimsLength = 3;
+        (*info)[1].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[1].dims[0] = 1;
+        (*info)[1].dims[1] = 3;
+        (*info)[1].dims[2] = 4;
+        return 0;
+    }
+    static int release(void* ptr) {
+        free(ptr);
+        return 0;
+    }
+};
+
+struct LibraryDemultiplexer_2 {
```

Comment:
```suggestion
struct LibraryDemultiplexerTwoInputsOneOutput {
```

Reply:
More important than "one output/two inputs" is that it matches following node *shape*. Adding this kind of description to structure name will decrease readability